webui/src/lib/onboarding/repoOnboardingService.ts

```diff
@@ -0,0 +1,119 @@
+};
+
+export const canUseRepoOnboarding = async (
+  repoId: string,
+  defaultBranch = "main"
+): Promise<boolean> => {
+  const promises = [];
+  promises.push(retention.setGCPolicyPreflight(repoId));
+  promises.push(branchProtectionRules.createRulePreflight(repoId));
+  promises.push(
+    objects.list(
+      repoId,
+      defaultBranch,
+      `${HOOKS_PATH}${generateRandomFileName()}`
+    )
+  );
+  const results = await Promise.allSettled(promises);
+  if (results.some((result) => result.status === "rejected" || !result.value)) {
+    return false;
+  }
+  return true;
+};
+
+export const getRepoOnboardingSteps = (
+  currentRepo: string,
+  defaultBranch: string,
+  objectStoreName: string,
+  navigate: NavigateFunction
+): OnBoardingSteps => [
+  {
+    id: "import-data",
+    title: "Import your existing data",
+    description: `Import your existing data from ${objectStoreName} to your repository by creating pointers to it (no copying of data involved)`,
+    cta: "Run import",
+    onClick: () =>
+      navigate(`/repositories/${currentRepo}/objects?importDialog=true`),
+    showStep: () => objectStoreName !== "local",
+    isCompleted: async () => {
+      const repoBranches = await branches.list(currentRepo);
+      // when the API client is typed, we can remove the eslint-disable
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return repoBranches.results.some((branch: any) =>
+        IMPORT_BRANCH_REGEX.test(branch.id)
+      );
+    },
+  },
+  {
+    id: "garbage-collection",
+    title: "Setup garbage collection",
+    description:
+      "Setup rules to automatically manage when data can be deleted from your repository",
+    cta: "Configure",
+    onClick: () => navigate(`/repositories/${currentRepo}/settings/retention`),
+    showStep: () => true,
+    isCompleted: async () => {
+      try {
+        await retention.getGCPolicy(currentRepo);
+        return true;
+      } catch (e) {
+        return false;
+      }
+    },
+  },
+  {
+    id: "branch-protection",
+    title: "Set up branch protection",
+    description:
+      "Add a quality gate to ensure your most important branches pass quality, compliance, or privacy checks",
+    cta: "Configure",
+    onClick: () => navigate(`/repositories/${currentRepo}/settings/branches`),
+    showStep: () => true,
+    isCompleted: async () => {
+      const rules = await branchProtectionRules.getRules(currentRepo);
+      return rules.length > 0;
+    },
+  },
+  {
+    id: "hooks",
+    title: "Set up hooks",
+    description:
+      "Create simple hooks to validate new data meets certain requirements: schema validation, metadata registration, and more.",
+    cta: "Configure",
+    onClick: () =>
+      window.open("https://docs.lakefs.io/hooks/overview.html", "_blank"),
+    showStep: () => true,
+    isCompleted: async () => {
+      const objectList = await objects.list(currentRepo, defaultBranch, "");
+      return objectList.results.some(
```

Comment:
This fails for a repository with many objects!  `objects.list` returns the first few objects, which might not include the hooks.  We need to check the specific hooks prefix (`_lakefs_actions/`) here.  But even that is not enough -- maybe I wrote the wrong hook object name, and nothing happened?

I would prefer to look at the actions, say verify that GET /repositories/{repository}/actions/runs returns a nonempty result.